api/tests/python/tests/check/check.py

```diff
@@ -0,0 +1,41 @@
+    { "name" : "issue269test_Issue_293_All",  "test": "radical;pseudoatom;stereo;query;overlap_atom;overlap_bond;rgroup;chirality;3d_coord", "expected": '''[{"code":6,"message":"Structure contains radicals","ids":[13,15,17]},{"code":14,"message":"Structure contains query features"}]'''},
+    ] 
+
+errors = ''
+for test in tests:
+    print(f"\nTEST: {test['name']}\n")
+    with open(joinPath(f"molecules/{test['name']}.mol"), 'r') as file:
+        molfile = file.read()
+    res = indigo.check2(molfile, test['test'])
+    if (res == test['expected']):
```

Comment:
You can follow the (quite interesting) approach used here in api python tests. 
Instead defining expected data and comparing with results inside the test, each test has its reference expected output (ref/**/test*.out) and the frame above compares the output produced by each test with this expected reference.